controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -134,50 +135,19 @@ public void createScope(final CreateScopeRequest createScopeRequest, final Secur
 
     private void authenticateAuthorize(String resourceName, AuthHandler.Permissions level) throws AuthException {
         if (pravegaAuthManager != null ) {
+            Map<String, String> map = null;
             List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
 
             if (authParams == null || authParams.isEmpty()) {
                 throw new AuthenticationException("Auth failed for " + resourceName);
             }
 
-            String credentials = authParams.get(0);
-            Preconditions.checkNotNull(credentials, "Credentials not specified in the parameters.");
+            String authHeader = authParams.get(0);
+            map = Arrays.stream(authHeader.split(",")).map(str -> str.split(":"))
+                                            .collect(Collectors.toMap(e -> e[0], e -> e[1]));
 
-            if (!pravegaAuthManager.authenticateAndAuthorize(resourceName, credentials, level)) {
```

Comment:
This whole block was not removed in the original PR #2385, why is it removed removed here?

Reply:
Apart from revert of #2385, some part of API changes introduced because of CR comments on 2824,2816 also needed to be changed. This is a change in `AuthHandler` API where we had introduced `Principal` concept instead of passing the parameters as a Map. That needed to be reverted as well.

Reply:
In which PR was this block changed? I'm looking at #2825 and I don't see changes to `StreamMetadataResourceImpl`.

Reply:
This was changed as part of this commit. https://github.com/pravega/pravega/commit/d35b5b99a6137e6c94648e80f9dbc8c64a72d766